dotnet/src/Connectors/Connectors.HuggingFace/Core/HuggingFaceEndpointProvider.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+
+namespace Microsoft.SemanticKernel.Connectors.HuggingFace.Core;
+
+internal sealed class HuggingFaceEndpointProvider : IEndpointProvider
+{
+    private readonly Uri? _endPoint;
+    private readonly string _separator;
+
+    /// <summary>
+    /// Initializes a new instance of the HuggingFace Endpoints class with the specified API key.
+    /// </summary>
+    /// <param name="endPoint">Endpoint url for Hugging Face Inference API</param>
+    public HuggingFaceEndpointProvider(Uri? endPoint)
+    {
+        if (endPoint is null)
+        {
+            endPoint = new Uri("https://api-inference.huggingface.co");
+        }
+
+        this._separator = endPoint.AbsolutePath.EndsWith("/", StringComparison.InvariantCulture) ? string.Empty : "/";
+        this._endPoint = endPoint;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the HuggingFace Endpoints class with the public API endpoint.
+    /// </summary>
+    /// <remarks>
+    /// Public API endpoint: https://api-inference.huggingface.co
+    /// </remarks>
+    public HuggingFaceEndpointProvider() : this(null)
+    {
+    }
+
+    public Uri GetTextGenerationEndpoint(string modelId)
+        => new($"{this._endPoint}{this._separator}models/{modelId}");
+
+    public Uri GetEmbeddingGenerationEndpoint(string modelId)
+        => new($"{this._endPoint}{this._separator}pipeline/feature-extraction/{modelId}");
```

Comment:
Can't we simply have the methods in the HuggingFaceClient class, use "https://api-inference.huggingface.co" as the default endpoint and override it with the value of BaseAddress, if it's set, in the provided HTTP client or the value of the endpoint parameter? This way, we can easily simplify the design by eliminating the need for the additional class.

Reply:
As the endpoint is a `Url` now, I cant use `new` as the default for the constructor, had to do this way. Any suggestion?

Reply:
EndPointProvider is a valid responsibility segregation, I can move this responsibility to the Client, sure, but I liked this approach from the Gemini impl which I used in Ollama's too.